projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-color.service.ts

```diff
@@ -22,6 +22,37 @@ export class DataLayerColorService {
     }
   }
 
+  chooseAnnotationColorsFromPalette(layer: DataLayer): void {
+    if(layer.annotations){
+      for (let annotation of layer.annotations) {
+        if (!this.getAnnotationColor(annotation)) {
+          const color = this.palette[this.nextColorIndex];
+          this.nextColorIndex = (this.nextColorIndex + 1) % this.palette.length;
+          this.setAnnotationColor(annotation, color);
+        }
+      }
+    }
```

Comment:
For now, I think I would prefer to leave the default light blue color for all annotations unless user chooses a different one. At least until we have a way to link annotation to a dataset and use matching colors.